src/nvim_helpers/main.go

```diff
@@ -1,36 +1,46 @@
 package nvim_helpers
 
 import (
+	"context"
+	"errors"
 	"fmt"
 	"strings"
 	"time"
 
 	"github.com/neovim/go-client/nvim"
 
-	"nvrh/src/context"
+	nvrh_context "nvrh/src/context"
 )
 
-func WaitForNvim(nvrhContext *context.NvrhContext) (*nvim.Nvim, error) {
+func WaitForNvim(ctx context.Context, nvrhContext *nvrh_context.NvrhContext) (*nvim.Nvim, error) {
```

Comment:
Honestly just the proper timeout changes here get enough of the "hangs while connecting to invalid host" cleared, just can't press ctrl-c to exit early without the context integration.